mw_url_rewrite.go

```diff
@@ -157,51 +153,64 @@ func urlRewrite(meta *apidef.URLRewriteMeta, r *http.Request) (string, error) {
 
 		// put url_rewrite path to context to be used in ResponseTransformMiddleware
 		ctxSetUrlRewritePath(r, meta.Path)
-
-		// matched?? Set the modified path
-		// return newpath, nil
 	}
 
+	newpath = replaceTykVariables(r, newpath, true)
+
+	return newpath, nil
+}
+
+const (
+	metaLabel    = "$tyk_meta."
+	contextLabel = "$tyk_context."
+)
+
+var dollarMatch = regexp.MustCompile(`\$tyk_context.([A-Za-z0-9_\-\.]+)`)
+var metaDollarMatch = regexp.MustCompile(`\$tyk_meta.([A-Za-z0-9_\-\.]+)`)
+
+func replaceTykVariables(r *http.Request, in string, escape bool) string {
 	contextData := ctxGetData(r)
 
-	dollarMatch := regexp.MustCompile(`\$tyk_context.([A-Za-z0-9_\-\.]+)`)
-	replace_slice := dollarMatch.FindAllStringSubmatch(rewriteToPath, -1)
-	for _, v := range replace_slice {
-		contextKey := strings.Replace(v[0], "$tyk_context.", "", 1)
+	if strings.Contains(in, contextLabel) {
```

Comment:
we could eliminate a couple of extra ops if move `contextData := ctxGetData(r)` inside this if